tests/components/konnected/test_panel.py

```diff
@@ -373,3 +373,163 @@ def mock_constructor(host, port, websession):
             ],
         }
     }
+
+
+async def test_default_options(hass, mock_panel):
+    """Test that we create a Konnected Panel and save the data."""
+    device_config = config_flow.CONFIG_ENTRY_SCHEMA(
+        {
+            "host": "1.2.3.4",
+            "port": 1234,
+            "id": "112233445566",
+            "model": "Konnected Pro",
+            "access_token": "11223344556677889900",
+            "default_options": config_flow.OPTIONS_SCHEMA(
+                {
+                    "io": {
+                        "1": "Binary Sensor",
+                        "2": "Binary Sensor",
+                        "3": "Binary Sensor",
+                        "4": "Digital Sensor",
+                        "5": "Digital Sensor",
+                        "6": "Switchable Output",
+                        "out": "Switchable Output",
+                    },
+                    "binary_sensors": [
+                        {"zone": "1", "type": "door"},
+                        {
+                            "zone": "2",
+                            "type": "window",
+                            "name": "winder",
+                            "inverse": True,
+                        },
+                        {"zone": "3", "type": "door"},
+                    ],
+                    "sensors": [
+                        {"zone": "4", "type": "dht"},
+                        {"zone": "5", "type": "ds18b20", "name": "temper"},
+                    ],
+                    "switches": [
+                        {
+                            "zone": "out",
+                            "name": "switcher",
+                            "activation": "low",
+                            "momentary": 50,
+                            "pause": 100,
+                            "repeat": 4,
+                        },
+                        {"zone": "6"},
+                    ],
+                }
+            ),
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected",
+        title="Konnected Alarm Panel",
+        data=device_config,
+        options={},
+    )
+    entry.add_to_hass(hass)
+    hass.data[panel.DOMAIN] = {
```

Comment:
It's better to set up the integration using the mock entry than to mock integration code. Whenever we mock integration code in tests of the integration, the tests become less robust.

Reply:
I understand.  The panel module is well isolated from the rest of the integration and HA architecture - it just needs to parse and act upon passed in config entry data (which the integration also defines).  

Bypassing the integration setup made it really clean to access, control, and assert behaviors based on the panel class methods.  I think these tests might be an exception where directly mocking some workings of the integration shouldn't add maintenance overhead.

Reply:
That we need to mock this is a sign that it's not isolated.

We don't really care about what the panel is doing. We just care about what the result in the core is upon different input that the panel forwards. It's better to set up the integration and assert core state.

Reply:
Or if the info is flowing the other way, we assert that the library is called correctly.

Reply:
ok - will fix